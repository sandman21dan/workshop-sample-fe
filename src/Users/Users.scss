.users {
  $table-colour: whitesmoke;
  // $table-colour: #f5cbb3;
  $table-radius: 4px;
  $table-padding: .8rem;

  box-shadow: 0px 3px 5px -5px rgba(0, 0, 0, 0.5);
  border-radius: 4px;
  overflow: hidden;

  &__row {
    background-color: $table-colour;

    &:nth-of-type(odd) {
      background-color: darken($table-colour, 3%);
    }
  }

  &__cell {
    padding: .3rem;

    &:first-of-type {
      padding-left: $table-padding;
    }

    &:last-of-type {
      padding-right: $table-padding;
    }

    & + & {
      padding-left: $table-padding;
    }
  }

  &__heading {
    background-color: darken($table-colour, 7%);

    & > * {
      text-align: left;
      padding-top: $table-padding;
      padding-bottom: $table-padding;
    }
  }
}
