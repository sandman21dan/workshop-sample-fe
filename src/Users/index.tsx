import * as React from 'react';

import { User } from '../api/User.type';

import './Users.scss';

const User: React.FunctionComponent<{ user: User }> = ({
  user,
}) => (
  <tr className="users__row">
    <td className="users__cell">{user.name}</td>
    <td className="users__cell">{user.lastName}</td>
    <td className="users__cell">{user.phone}</td>
  </tr>
);

const Users: React.FunctionComponent<{ users: User[] }> = ({
  users,
}) => (
  <table className="users" cellSpacing="0">
    <tr className="users__heading">
      <th className="users__cell">Name</th>
      <th className="users__cell">Last name</th>
      <th className="users__cell">Phone Number</th>
    </tr>
    {users.map((user, idx) => (<User user={user} key={idx} />))}
  </table>
);

export default Users;
