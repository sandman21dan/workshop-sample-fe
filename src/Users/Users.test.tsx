import * as React from 'react';

import Users from './';
import { User } from '../api/User.type';
import { shallow } from 'enzyme';

const userList: User[] = [
  {
    id: 'GUUID-1',
    name: 'John',
    lastName: 'Doe',
    phone: '0777665544',
  },
  {
    id: 'GUUID-2',
    name: 'Johnathan',
    lastName: 'Donald',
    phone: '0777665543',
  }
];

it('renders correctly with no users', () => {
  const component = shallow(<Users users={[]} />);

  expect(component).toMatchSnapshot();
});

it('renders correctly with users', () => {
  const component = shallow(<Users users={userList} />);

  expect(component).toMatchSnapshot();
});
