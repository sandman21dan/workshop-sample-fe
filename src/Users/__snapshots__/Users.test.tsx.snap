// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no users 1`] = `
<table
  cellSpacing="0"
  className="users"
>
  <tr
    className="users__heading"
  >
    <th
      className="users__cell"
    >
      Name
    </th>
    <th
      className="users__cell"
    >
      Last name
    </th>
    <th
      className="users__cell"
    >
      Phone Number
    </th>
  </tr>
</table>
`;

exports[`renders correctly with users 1`] = `
<table
  cellSpacing="0"
  className="users"
>
  <tr
    className="users__heading"
  >
    <th
      className="users__cell"
    >
      Name
    </th>
    <th
      className="users__cell"
    >
      Last name
    </th>
    <th
      className="users__cell"
    >
      Phone Number
    </th>
  </tr>
  <User
    key="0"
    user={
      Object {
        "id": "GUUID-1",
        "lastName": "Doe",
        "name": "John",
        "phone": "0777665544",
      }
    }
  />
  <User
    key="1"
    user={
      Object {
        "id": "GUUID-2",
        "lastName": "Donald",
        "name": "Johnathan",
        "phone": "0777665543",
      }
    }
  />
</table>
`;
